ddtrace/tracer/tracer_test.go

```diff
@@ -946,6 +946,46 @@ func TestTracerFlush(t *testing.T) {
 	})
 }
 
+func TestTracerReportsHostname(t *testing.T) {
+	t.Run("enabled", func(t *testing.T) {
+		os.Setenv("DD_TRACE_REPORT_HOSTNAME", "true")
+		defer os.Unsetenv("DD_TRACE_REPORT_HOSTNAME")
+
+		tracer, _, stop := startTestTracer()
+		defer stop()
+
+		root := tracer.StartSpan("root").(*span)
+		child := tracer.StartSpan("child", ChildOf(root.Context())).(*span)
+		child.Finish()
+		root.Finish()
+
+		assert := assert.New(t)
+
+		name, ok := root.Meta[keyHostname]
+		assert.True(ok)
+		assert.Equal(name, tracer.hostname)
+
+		_, ok = child.Meta[keyHostname]
+		assert.False(ok)
+	})
+	t.Run("not enabled", func(t *testing.T) {
```

Comment:
It would be nice to add a space above this. Also, it might be nicer to use `disabled` as the test name instead of `not enabled`. Using space separated words in sub-test names make them look a bit weird when they're being run with `-v`